### PR TITLE
Added Utils namespace with WorldZone class

### PR DESCRIPTION
**Pull request by LukeD7**
The WorldZone class keeps a list of the known world zone strings from in game (eg. airp, canny, grapes etc).

This class creates a WorldZone object from it and stores it's game name (as above) and it's friendly name (Grapes becomes Grapeseed etc). Each "zone" is also attached to a County enum so it is clearer which county each zone is in.

### CODE REVIEW

File: Utils/WorldZone.cs [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+            GameName = gameName;
+            RealAreaName = areaName;
+            this.county = county;
+        }
+
+
+        internal static WorldZone GetZoneAtPosition(Vector3 position)
+        {
+            if (Zones.Count <= 0)
+                RegisterWorldZones();
```
Comment (Salzian) [inline Utils/WorldZone.cs:41]: Why is this not initialised in the constructor?

File: Utils/WorldZone.cs [hunk 2]
```diff
@@ -0,0 +1,162 @@
+﻿using System.Collections.Generic;
+using System.Text.RegularExpressions;
+using CitizenFX.Core;
+using CitizenFX.Core.Native;
+
+namespace FivePD-API.Utils
```
Comment (Salzian) [inline Utils/WorldZone.cs:6]:
```suggestion
namespace FivePD.API.Utils
```
As per Microsoft's namespace conventions I would drop the hyphen. https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/names-of-namespaces

File: Utils/WorldZone.cs [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+            Zones.Add(new WorldZone("VCana", "Vespucci Canals", EWorldZoneCounty.LosSantos));
+            Zones.Add(new WorldZone("Vesp", "Vespucci", EWorldZoneCounty.LosSantos));
+            Zones.Add(new WorldZone("Vine", "Vinewood", EWorldZoneCounty.LosSantos));
+            Zones.Add(new WorldZone("WVine", "West Vinewood", EWorldZoneCounty.LosSantos));
+            Zones.Add(new WorldZone("WindF", "Ron Alt. Wind Farm", EWorldZoneCounty.LosSantosCounty));
+            Zones.Add(new WorldZone("ZP_ORT", "Port of South LS", EWorldZoneCounty.LosSantos));
+            Zones.Add(new WorldZone("Zancudo", "Zancudo River", EWorldZoneCounty.BlaineCounty));
+            Zones.Add(new WorldZone("zQ_UAR", "Davis Quartz Quarry", EWorldZoneCounty.LosSantosCounty));
+            Zones.Add(new WorldZone("SanAnd", "San Andreas", EWorldZoneCounty.SanAndreas));
+        }
```
Comment (Salzian) [inline Utils/WorldZone.cs:150]: Could all be done in the constructor saving one if-call every time calling GetZoneAtPosition as well as possible lag at first call due to List population.

**Review by Salzian — COMMENTED** [review]
+1 for separate file

File: Utils/WorldZone.cs (hunk 2 above)
Comment (LukeD7) [reply Utils/WorldZone.cs:6]: I've simply mirrored the current naming convention from the API. This is merely a placeholder until a set up is defined by the devs. But yes, I do agree.

File: Utils/WorldZone.cs (hunk 3 above)
Comment (LukeD7) [reply Utils/WorldZone.cs:150]: Never experienced any lag using this in singleplayer, even in demanding settings.

Comment (Salzian) [reply Utils/WorldZone.cs:150]: Still does not seem to me as a good practice and I don't see any reason to do it like this.

File: Utils/WorldZone.cs (hunk 1 above)
Comment (LukeD7) [reply Utils/WorldZone.cs:41]: No need to add it to the c.tor, saves populating a list without reason. If this is never used by the dev making use of the API then I see no reason to populate a list. Rather populate it on it's first call. This is also a static call, the c.tor creates each of the zones not the list.

Comment (Salzian) [reply Utils/WorldZone.cs:41]: Per C# language specification, a static field is being populated prior to using the class for the first time automatically on an implementation-level, therefore making this unnecessary. If someone is never using/referencing this class, the static field will never be populated. https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/language-specification/variables#static-variables

Comment (Salzian) [reply Utils/WorldZone.cs:41]: I would just initialise the variable inline and let C# decide when to allocate memory.

**Comment by KDani-99** [comment]
Added in 1.0.
Thanks.

**Pull request closed without merging**